scripts/apps/desks/services/DesksFactory.ts

```diff
@@ -284,6 +284,7 @@ export function DesksFactory($q, api, preferencesService, userList, notify,
         },
         fetchDeskById: function(Id) {
             return api.desks.getById(Id).then((_desk) => {
+                _desk.desk_metadata = _desk.desk_metadata ?? {};
```

Comment:
On the other hand, if the property is optional, we could mark it so in the types and access using `desk?.desk_metadata` and if it's not optional, server could always return an empty object so we don't need to override it here. Otherwise, desks will also be accessible directly via the endpoint in any other place in the code and will not have that property,

Reply:
it's optional, we only need it really set on that ui component in desk settings, but there it's accessed from directive which is expecting it to be set (this component is used for different types and is not desk specific).

Reply:
> we only need it really set for ui component in desk settings

Is it a real property that desks really have and store it in the database or is it front-end-only pseudo-property like `IArticle['selected']` that should not be saved to the database?

If so, I would then keep it in that UI component and would not spread this bad practice to other parts of the code.

Reply:
it's real property and gets stored

Reply:
I would then set that as optional in the interface and use optional chaining for accessing data. For angular templates optional chaining will probably not work, so we could initialize a default.

The current solution is fine, but I would add the property as optional to the interface so developers know that optional chaining has to be used in other places.